accepted/2022/simplify-rid-model.md

```diff
@@ -0,0 +1,249 @@
+
+For .NET 8, we'll continue with the model. However, we will no longer be able to use [CentOS 7 (due to EOL)](https://wiki.centos.org/About/Product), but will need to adopt another distro that provides us with an old enough glibc version such that we can offer broad compatibility.
+
+For .NET 9, we'll likely drop support for Windows 10 and instead target the Windows 11 CRT.
+
+As part of investigating this topic, we noticed the [libc compatibility model that TensorFlow uses](https://github.com/tensorflow/tensorflow/blob/f3963e82f21c9d094503568699877655f9dac508/tensorflow/tools/ci_build/devtoolset/build_devtoolset.sh#L48-L57). That model enables the use of a modern OS with an artificially old libc. We also saw that Python folks are doing something similar with their [`manylinux` approach but with docker](https://github.com/pypa/manylinux).
+
+This in turn made us realize that our [`build-tools-prereqs` Docker images](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/centos/7/Dockerfile) are very similar to the Python approach. We don't need a new contract like `manylinux2014` since we can rely on the contract changing with each major .NET version. We need augment our build-tools-prereq scheme to also include an old glibc version, once CentOS 7 is EOL.
+
+The biggest difference with our `build-tools-prereqs` images is that they are not intended for others to use, while the `manylinux` images are intended as a general community artifact. There are two primary cases where extending the use of the `build-tools-prereqs` images would be useful: enabling others to produce a .NET build with the same compatibility reach, and enabling NuGet authors to build native dependencies with matching compatibility as the underlying runtime. Addressing that topic is outside the scope of this document. This context is included to help frame how the TensorFlow, Python, and .NET solutions compare, and to inspire future conversation. 
+
+Note: It appears that some other folks have been [reacting to CentOS 7 not being a viable compilation target](https://gist.github.com/wagenet/35adca1a032cec2999d47b6c40aa45b1) for much longer.
+
+## Legal RIDs
+
+RIDs are defined by legal combinations of operating system, chip architecture, and C-runtime. We support both triplets (primarily native code) and singles (only processor-agnostic managed code).
+
+A given RID is used to describe:
+
+- This system supports this RID, per the .NET host.
+- This code will run on / requires a system that supports this RID, per the author.
+
+The following table describes RIDs supported by the .NET host. It is exhaustive, per the hosts provided by Microsoft.
+
+| RID        | Description of scope |
+|------------|----------------------|
+| unix       | All Unix-based OSes (macOS and Linux), versions, and architecture builds. |
+| win        | All Windows versions and architecture builds. |
+| linux-arm32| All Linux glibc-based distros, for Arm32. |
```

Comment:
Are there any plans to add [`linux-armv6`](https://github.com/dotnet/runtime/pull/62594) here?

Reply:
Ya. I'll add it. We can decide whether it fully makes sense later.